speechbrain/core.py

```diff
@@ -658,8 +322,9 @@ def __init__(  # noqa: C901
         self.optimizers_dict = None
         self.opt_class = opt_class
         self.checkpointer = checkpointer
+        _,self.run_opt_defaults,_ = RunOptions.from_command_line_args()
 
-        for arg, default in run_opt_defaults.items():
+        for arg, default in self.run_opt_defaults.items():
             if run_opts is not None and arg in run_opts:
                 if hparams is not None and arg in hparams:
                     logger.info(
```

Comment:
The defaults are all handled by the class now. So this shouldn't have to call `from_command_line_args()` at all, which can be done in the recipes. Instead, the class should just take a `RunOptions` object.